lib/ansible/modules/windows/win_data_deduplication.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: 2019, rnsc(@rnsc) <github@rnsc.be>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: win_data_deduplication
+version_added: "2.9"
+short_description: Module to enable Data Deduplication on a volume.
+description:
+- This module can be used to enable Data Deduplication on a Windows volume.
+- You have to have the FS-Data-Deduplication feature installed.
+- This module doesn't support check_mode due to a lack of WhatIf support.
+options:
+  drive_letter:
+    description:
+    - Windows drive letter on which to enable data deduplication.
+    required: yes
+    type: str
+  state:
+    description:
+    - Wether to enable or disable data deduplication on the selected volume.
+    required: yes
+    default: enabled
+    type: str
+    choices: [ enabled, disabled ]
+  settings:
```

Comment:
You should be doing a dictionary instead of list.

In fact you should be setting these known options as actual documented module options i.e. have `optimize_in_use_files: yes/no`, `no_compress`, `verify`, minimum_file_age_days`. Having an open list/dict of options means user's have to go and read other docs to see what is available and work from there. If we document the options explicitly then they can read our docs and we don't have to handle arbitrary input.

You don't need to implement all the options straight away, maybe just what you need right now.